crates/bevy_pbr/src/material.rs

```diff
@@ -582,28 +583,121 @@ pub const fn screen_space_specular_transmission_pipeline_key(
     }
 }
 
+/// Fills the [`RenderMaterialInstances`] and [`RenderMeshMaterialIds`]
+/// resources from the meshes in the scene.
 fn extract_mesh_materials<M: Material>(
     mut material_instances: ResMut<RenderMaterialInstances<M>>,
     mut material_ids: ResMut<RenderMeshMaterialIds>,
     mut material_bind_group_allocator: ResMut<MaterialBindGroupAllocator<M>>,
-    query: Extract<Query<(Entity, &ViewVisibility, &MeshMaterial3d<M>)>>,
+    changed_meshes_query: Extract<
+        Query<
+            (Entity, &ViewVisibility, &MeshMaterial3d<M>),
+            Or<(Changed<ViewVisibility>, Changed<MeshMaterial3d<M>>)>,
```

Comment:
Question on this: MeshMaterial3d might not change, but the asset pointed to by its handle might change. Do we need to handle that here? I don't have the context of what extract is doing here anymore.

Reply:
We don't need to handle that here because the only thing `extract_mesh_materials` needs to do is to allocate within the slabs and store the resulting texture *indices*. If the material changes, then whatever textures (and other bindings) the material has will be updated in-place within the slabs, so the indices won't change, so this system doesn't need to do anything.